core/src/main/java/org/apache/iceberg/io/WriteResult.java

```diff
@@ -76,6 +76,11 @@ public Builder add(WriteResult result) {
       return this;
     }
 
+    public Builder add(Iterable<WriteResult> results) {
```

Comment:
Typically, we would follow the Java collection convention and use `addAll`.

Reply:
OK,  rename it to `addAll` sound great to me.